test/functional/log.rb

```diff
@@ -201,6 +201,13 @@ def assert_logged(expected_string, *args)
           end
         end
 
+        it 'prefers explicit information over the context' do
+          log.with_contextual_info(omg: 'wtf') do
+            log.info("message", omg: 'ponies')
```

Comment:
...and the bug was nondeterministic because the `chalk-log` test suite mutates global state in a really awful way, and we were getting the result of the `adds contextual information` test above instead of our test